### DISCUSSION TITLE
Procedural DNA styling

### DISCUSSION THREAD

**Discussion by janissch-unibe** [BradyAJohnston/MolecularNodes#138] [Q&A] [answered]
Hi! I am working on a video explaining the role of RNA in new medication techniques for my university and have a few questions regarding the procedural DNA feature.

- Can I somehow only render one strand of the procedural DNA in order to make procedural RNA?
- Is there currently a way to apply a nucleic ribbon style like in the image also for procedurally generated DNA?

![grafik](https://user-images.githubusercontent.com/122604352/212289289-daea8c84-838b-4035-8ef1-11be61d08e68.png)

Thanks!
Janis

**Comment by BradyAJohnston** [comment, marked as answer]
Hello! Thanks for the question. Yes you can definitely render just one strand. The outputted DNA should have an attribute `chain_number` which will be either 1 or 2, so you can delete all of the geometry from one of the strands leaving the other.

For the more simplified procedural DNA look, you can achieve that by using the curve that is outputted from the DNA node, and use curve to mesh. You can also build your own version of the DNA node by following these tutorials I have made previously: https://www.youtube.com/watch?v=m7bTEABoAzo&t=0s

**Reply by janissch-unibe** [reply]
Perfect, thank you!

**Reply by BradyAJohnston** [reply]
Would love to see what you come up with as well. Please do share when you are done!

**Reply by karlduderstadt** [reply]
I had the same question but due to my lack of blender knowledge I am having trouble implementing your answer with MN 4.2.5. I wanted to remove one chain and tried the following geometry node layout but depending on the `chain_id` either both strands appear or no strands appear. Can you see what I am doing wrong?

<img width="992" alt="Screenshot 2024-09-12 at 14 30 09" src="https://github.com/user-attachments/assets/fac6868f-3c53-4d6e-b88c-327a6826b4fd">

I was also wondering how to get the procedurally generated DNA to have the same cartoon ribbon style as DNA imported using a PDB. To do this you suggest using the Helix Curve and converting to mesh. How do you make it the same thickness as seen in the cartoon? Also, the bases don't have the same style and are shifted when just joining the geometry output from the DNA double helix node. Do you have any other hints of how to get the cartoon style? I am eager to contribute to the effort of adding a procedural DNA cartoon node style to MN if I can help in anyway.

It is amazing what you have made possible with this library. Thank you very much for working on this and sharing it!

**Reply by karlduderstadt** [reply]
Oh I just found this discussion (https://github.com/BradyAJohnston/MolecularNodes/discussions/151). That addressed my first question. Sorry, I am not sure how I didn't see that before. I am still curious the best way forward to achieve the cartoon style of procedural DNA. I am watching your earlier DNA video (that you linked) that explains how to make DNA from geometry nodes, but it would be nice to instead add a different style node that goes after MN_dna_bases, so one could switch between styles later if desired.
